week-3/Homework/mandatory/7-password-validator.js

```diff
@@ -1,69 +1,78 @@
 contains a valid password (see below for password criterias) and return 
 new array with true or false booleans.
-
 Passwords must 
 - Have at least 5 characters.
 - Have English uppercase letters (A-Z)
 - Have English lowercase letters (a-z)
 - Have numbers (0-9)
 - Have non-alphanumeric symbols ("!", "#", "$", "%", ".")
-
 Passwords must not be any previous password in the passwords array. 
-
 Example 1:
 PreviousPassword = ["fhD8!yrjj", "ttkTu.wer3", "dvyyeyY!5", "qwbfj76%", "tytT3729."];
-
 Expected Result:
 PasswordValidationResult=  [false, false, false, false, true]
-
 */
 
-function validatePasswords(passwords) {
+let allUpper = /[A-Z]/;
+let allLower = /[a-z]/;
+let allNumbers = /[0-9]/;
+let allSymbols = /[!#$%.]/;
 
+function validatePasswords(passwords) {
+  let filterPasswords = passwords.map(function(checkPassword) {
+    if (
+      checkPassword.length >= 5 &&
+      allUpper.test(checkPassword) &&
+      allLower.test(checkPassword) &&
+      allNumbers.test(checkPassword) &&
+      allSymbols.test(checkPassword)
+    ) {
+      return true;
+    } else {
+      return false;
+    }
+  });
+  console.log(filterPasswords);
+  return filterPasswords;
 }
 
 /* ======= TESTS - DO NOT MODIFY ===== */
 
-const passwords1 = ["Se%5", "TktE.TJTU", "384HsHF", "dvyyeyY!5", "tryT3729."]
-const passwords2 = ["StUFf27", "PlEnty", "Jai.33" "shajsaUA**&&", "PlEnty"]
+const passwords1 = ["Se%5", "TktE.TJTU", "384#HsHF", "dvyyeyy!5", "tryT3729"];
+const passwords2 = ["StUFf27%", "Pl3nty!", "Jai33", "shajsaUA**&&", "Pl3nty!"];
 
 function arraysEqual(a, b) {
-    if (a === b) return true;
-    if (a == null || b == null) return false;
-    if (a.length != b.length) return false;
-  
-    for (let i = 0; i < a.length; ++i) {
-      if (a[i] !== b[i]) return false;
-    }
-  
-    return true;
+  if (a === b) return true;
+  if (a == null || b == null) return false;
+  if (a.length != b.length) return false;
+
+  for (let i = 0; i < a.length; ++i) {
+    if (a[i] !== b[i]) return false;
+  }
+
+  return true;
 }
 
 function test(test_name, expr) {
-    let status;
-    if (expr) {
-      status = "PASSED";
-    } else {
-      status = "FAILED";
-    }
-  
-    console.log(`${test_name}: ${status}`);
+  let status;
+  if (expr) {
+    status = "PASSED";
+  } else {
+    status = "FAILED";
+  }
+
+  console.log(`${test_name}: ${status}`);
 }
 
 test(
-   "validatePasswords function works - case 1",
-   arraysEqual(
-      validatePasswords(passwords1), [false, false, true, false, false]
-   )
- );
+  "validatePasswords function works - case 1",
+  arraysEqual(validatePasswords(passwords1), [false, false, true, false, false])
+);
 
- test(
-   "validatePasswords function works - case 2",
-   arraysEqual(
-      validatePasswords(passwords2), [true, true, false, false, false]
-   )
- );
\ No newline at end of file
+test(
+  "validatePasswords function works - case 2",
+  arraysEqual(validatePasswords(passwords2), [true, true, false, false, true])
```

Comment:
Hey Alex, the exercise looks good but you shouldn't change the test as it works as expected. From the exercise text:

> Passwords must not be any previous password in the passwords array.

Reply:
Hy Francesc,
I.m gone remake it. Thank you!